backend/matching-service/app/src/matching/matching.gateway.ts

```diff
@@ -0,0 +1,42 @@
+import { v4 as uuidv4 } from "uuid";
+import { Preferences, UserSocket, WaitingSocket } from "./matching.interfaces";
+
+export class MatchingGateway {
+  public roomIdToSockets: Map<string, UserSocket[]> = new Map();
+  public waiting: WaitingSocket[] = [];
+  private roomIdToLinkWaiting: Map<string, UserSocket> = new Map();
+
+  constructor() {}
+
+  createRoom(socket: any) {
+    const roomId = uuidv4();
+    socket.join(roomId);
+    this.waiting.push({
+      roomId: uuidv4(),
+      // Hard code one existing user for testing purposes
+      preferences: {
+        difficulty: ["Medium"],
+        category: ["Array"],
+      },
+      userSocket: { user: 1, socket },
+    });
+  }
+
+  joinRandomRoom(
+    user: UserSocket,
+    preferences: Preferences,
+    socket: any
+  ): string {
+    let roomId: string = "";
+    this.waiting.forEach((room) => {
+      if (JSON.stringify(room.preferences) === JSON.stringify(preferences)) {
```

Comment:
correct me if i'm wrong but this logic seems like it would cause problems if multiple preferences are found (i.e. the user might 'join' multiple rooms at once)

Reply:
oops my fault fixed in 0519d50cf90626283519b5841b718fed461df478 by just returning when found

Reply:
oh adding a `return` doesn't break the `forEach` loop like the way you're intending 

> There is no way to stop or break a forEach() loop other than by throwing an exception. (https://medium.com/front-end-weekly/3-things-you-didnt-know-about-the-foreach-loop-in-js-ff02cec465b1#:~:text=From%20the%20official%20MDN%20docs,method%20is%20the%20wrong%20tool.)

if you want to terminate the loop after finding a match, i think you should use a `for` loop instead